### ISSUE TITLE
Option to disable report output

### ISSUE THREAD

**Issue by donutbrew** [OpenGene/fastp#201]
Forgive me if this is already somewhere, but it would be great to have an option not to generate the trimming reports. Currently I tack on ` --json /dev/null --html /dev/null`

**Comment by mschilli87** [comment]
@donutbrew 
> Currently I tack on `--json /dev/null --html /dev/null`

Just out of curiosity: How would native `--no-json`/`--no-html` option be any better?

If it's just about the typing, how about `alias fpnr="fastp --json /dev/null --html /dev/null"`? :wink:

**Comment by donutbrew** [comment]
I actually think it would be a better default to have the report generation off, rather than default to a single filename which conflicts with parallel fastp processes. The user should be required to explicitly give a filename through `--json <file>` / `--html <file>` so that unwanted data is not generated and that _wanted_ data is not overwritten.

**Comment by mschilli87** [comment]
@donutbrew:
While I get your angle and I might even agree if we were discussing the behaviour of our own (yet unreleased!) software, I think at this point the number of scripts/Snakefiles broken that rely on the old behaviour outweigh the potential advantages.
I this case, the best solution IMHO would be to make sure `fastp` allows to specify the same parameters repeatedly, with the last instance taking precedence. I didn't check if that's the case but if not it could be changed without affecting established workflows by breaking backwards compatibility. Then, you could declare `alias fastp="fastp --json /dev/null --html /dev/null"` to get the default behaviour you want.

**Comment by sfchen** [comment]
I completely agree with @mschilli87 

For a tool that has been deployed widely, the first priority is to keep the backwards compatibility.

Remember that, although fastp offers many options to customize your processing, it is designed to work well with minimal options. That's why I assigned default filenames for JSON/HTML reports.

Output reports to `/dev/null` may be not the graceful solution, but is a good work around.

**Comment by Kasuax** [comment]
I have to agree, I would like to promote the latest flyway to our production environment but I cannot disable or turn off reports that continue to fail generating so we're going to have to rollback.  The reports are great but backwards compatibility should have been considered here.
